server/models/Users.js

```diff
@@ -96,12 +106,25 @@ class Users {
   }
 
   removeUser(username, callback) {
-    this.db.remove({username: username}, {}, (err, numRemoved) => {
+    this.db.findOne({username: username}).exec((err, user) => {
       if (err) {
         return callback(null, err);
       }
 
-      return callback({username: username});
+      // Username not found.
+      if (user == null) {
+        return callback(null, user);
+      }
+
+      this.db.remove({username: username}, {}, (err, numRemoved) => {
+        if (err) {
+          return callback(null, err);
+        }
+
+        fs.removeSync(`${config.dbPath}${user._id}/`);
```

Comment:
This should be asynchronous.

Reply:
totally asynchronous or waiting anyway for the method callback?
also, probably we should delete the directory first and then remove the user from the db. so if any error occurs we end up in a probably recoverable status